internal/identity/group/group.go

```diff
@@ -1,5 +1,307 @@
 package group
 
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/identity/entity"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
 const (
-	LookupPath = "identity/lookup/group"
+	LookupPath        = "identity/lookup/group"
+	IdentityGroupPath = "/identity/group"
+
+	GroupResourceType = iota
+	EntityResourceType
 )
+
+func IdentityGroupIDPath(id string) string {
+	return fmt.Sprintf("%s/id/%s", IdentityGroupPath, id)
+}
+
+// ReadIdentityGroup may return `nil` for the IdentityGroup if it does not exist
+func ReadIdentityGroup(client *api.Client, groupID string, retry bool) (*api.Secret, error) {
+	path := IdentityGroupIDPath(groupID)
+	log.Printf("[DEBUG] Reading IdentityGroup %s from %q", groupID, path)
+
+	return entity.ReadEntity(client, path, retry)
+}
+
+func IsIdentityNotFoundError(err error) bool {
+	return err != nil && errors.Is(err, entity.ErrEntityNotFound)
+}
+
+func getFieldFromResourceType(resourceType int) string {
+	var ret string
+	switch resourceType {
+	case GroupResourceType:
+		ret = consts.FieldMemberGroupIDs
+	case EntityResourceType:
+		ret = consts.FieldMemberEntityIDs
+	}
```

Comment:
Probably want to return an error for the unknown type.

Reply:
Fixed in 21f55a6